pymeasure/instruments/agilent/agilentB1500.py

```diff
@@ -155,6 +157,37 @@ def initialize_all_smus(self):
                 )
                 i += 1
 
+    def initialize_spgu(self, channel, name):
+        """Initializes SPGU instance by calling :class:`.SPGU`.
+
+        :param channel: SPGU channel
+        :type channel: int
+        :param spgu_type: SPGU type, e.g. ``'HRSPGU'``
+        :type spgu_type: str
+        :param name: SPGU name for pymeasure (data output etc.)
+        :type name: str
+        :return: SPGU instance
+        :rtype: :class:`.SPGU`
+        """
+        if channel in (list(range(101, 1101, 100)) + list(range(102, 1102, 100))):
+            channel = int(str(channel))
+        self._spgu_names[channel] = name
+        spgu_reference = SPGU(self, channel, name)
+        self._spgu_references[channel] = spgu_reference
+        return spgu_reference
+
+    def initialize_all_spgus(self):
+        """Initialize all SPGUs by querying available modules and creating
+        a SPGU class instance for each.
+        SPGUs are accessible via attributes ``.spgu1`` etc.
+        """
+        modules = self.query_modules()
+        i = 1
+        for channel, module_type in modules.items():
+            if module_type == "SPGU":
+                setattr(self, "spgu" + str(i), self.initialize_spgu(channel, "SPGU" + str(i)))
+                i += 1
+
     def pause(self, pause_seconds):
         """Pauses Command Execution for given time in seconds (``PA``)
```

Comment:
Typically, we do not tell in a docstring, which command is sent to the device.
But to keep consistency in this device, we can keep them in.